### PR TITLE
Feature/ret 1641 hide function

### PR DESCRIPTION
**Pull request by alig01**
Creating a map to store the filter values, so i don't need to load them over and over again.

### CODE REVIEW

File: src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java [hunk 1]
```diff
@@ -56,4 +60,25 @@ private SearchFilterFiles() {}
 			return Collections.emptyList();
 		}
 	}
+
+	public static Map<String, List<Filter>> getFilterMap() {
+		return filterMap != null ? filterMap : initTheFilterMap();
+	}
+
+	public static Map<String, List<Filter>> initTheFilterMap() {
```
[the remark this answers is not in the document]
Comment (beatngu13) [reply src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java]: If we can immediately initialize the map and the method can be private, I would suggest something like:

```java
private static final Map<String, List<Filter>> filterMap = initFilterMap();

// …

private static Map<String, List<Filter>> initFilterMap() {
	return getAllFilterFiles().stream() //
			.collect( Collectors.toMap( //
					filter -> filter.getFileName().toString(), //
					SearchFilterFiles::toFilterList ) );
}

private static List<Filter> toFilterList( final Path filter ) {
	try ( final Stream<String> filterFileLines = Files.lines( filter ) ) {
		return Loaders.load( filterFileLines ) //
				.filter( Filter.class::isInstance ) //
				.map( Filter.class::cast )//
				.collect( Collectors.toList() );
	} catch ( final IOException e ) {
		log.error( "Could not load '{}' file.", filter, e );
		return Collections.emptyList();
	}
}
```

**Review by beatngu13 — CHANGES_REQUESTED** [review]

File: src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java [hunk 2]
```diff
@@ -56,4 +60,25 @@ private SearchFilterFiles() {}
 			return Collections.emptyList();
 		}
 	}
+
+	public static Map<String, List<Filter>> getFilterMap() {
+		return filterMap != null ? filterMap : initTheFilterMap();
```
[the remark this answers is not in the document]
Comment (alig01) [reply src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java]: No, there was no special reason. nonetheless I have moved the variable in to review

File: src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (alig01) [reply src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java]: at least one has to be public so I can initialize the checkboxes and apply the filters.

File: src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java [hunk 3]
```diff
@@ -56,4 +58,23 @@ private SearchFilterFiles() {}
 			return Collections.emptyList();
 		}
 	}
+
+	public static Map<String, List<Filter>> loadFilterMap() {
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java]: I think, that this should not be part of the API. Sure, loading a `Filter` should be, but collecting the name should not. If we take a look at the MVC-Paradigma and its derivatives, the Model should never know how things are displayed (e.g. color, localization). However, it can be implicitly define things like name if it is part of the Model (e.g. `NamedFilter` - which I would not do).

Either way, Maps are a abstract Containers featuring easy key-value access. If you need the name and must have it here, we should not use a `Map` if we need access to the key (i.e. its entries), instead a List with a more descriptive Object.

Furthermore, I would not use a Container as such, just load a single `Filter` based on a `Path`, so that one can choose, which filters to load.

So to sum up: If we would have a method `NamedFilter loadNamedFilter(Path)` where we specifically pass the Path, what is hindering us in the GUI to just get the file name (under the assumption that we value the file name as actual name)? **Thus I would just do a method to load a `Filter` and let the GUI decide its name**, unless we decide that the `Filter` should not have its name based on the file name but instead on the file's contents (e.g. a line #name positioning).

Lastly, loading a filter should not be part of `SearchFilterFiles` since the class does not suggest that it can load a filter.

**Review by beatngu13 — APPROVED** [review]
